src/main/scala/io/opentargets/etl/backend/drug_beta/DrugBeta.scala

```diff
@@ -0,0 +1,115 @@
+      "gene" -> IOResourceConfig(common.inputs.target.format, common.inputs.target.path),
+      "evidence" -> IOResourceConfig(common.inputs.evidence.format, common.inputs.evidence.path)
+    )
+
+    val inputDataFrames = Helpers.readFrom(mappedInputs)
+
+    // raw input dataframes
+    lazy val moleculeDf: DataFrame = inputDataFrames("molecule")
+    lazy val mechanismDf: DataFrame = inputDataFrames("mechanism")
+    lazy val indicationDf: DataFrame = inputDataFrames("indication")
+    lazy val targetDf: DataFrame = inputDataFrames("target")
+    lazy val geneDf: DataFrame = inputDataFrames("gene")
+    lazy val drugbankData: DataFrame = inputDataFrames("drugbank")
+      .withColumnRenamed("From src:'1'", "id")
+      .withColumnRenamed("To src:'2'", "drugbank_id")
+    lazy val efoDf: DataFrame = inputDataFrames("efo")
+    lazy val evidenceDf: DataFrame = inputDataFrames("evidence")
+
+    // processed dataframes
+    logger.info("Raw inputs for Drug beta loaded.")
+    logger.info("Processing Drug beta transformations.")
+    val molecule = new Molecule(moleculeDf, drugbankData)
+    val indications = new Indication(indicationDf, efoDf)
+    val mechanismOfAction = new MechanismOfAction(mechanismDf, targetDf, geneDf)
+
+    val moleculeProcessedDf = molecule.processMolecules
+    val indicationProcessedDf = indications.processIndications
+    val mechanismOfActionProcessedDf = mechanismOfAction.processMechanismOfAction
+    val targetsAndDiseasesDf =
+      DrugCommon.getUniqTargetsAndDiseasesPerDrugId(evidenceDf).withColumnRenamed("drug_id", "id")
+    printDetailedLogging(moleculeProcessedDf,
+                         indicationProcessedDf,
+                         mechanismOfActionProcessedDf,
+                         targetsAndDiseasesDf)
+
+    logger.info(
+      "Joining molecules, indications, mechanisms of action, and target and disease linkages.")
+    // using inner joins as we don't want molecules that have no indications and mechanisms of action.
+    val drugDf: DataFrame = moleculeProcessedDf
+      .join(indicationProcessedDf, Seq("id"))
+      .join(mechanismOfActionProcessedDf, Seq("id"), "left_outer")
+      .join(targetsAndDiseasesDf, Seq("id"), "left_outer")
+      .transform(addDescription)
+
+    val outputs = Seq("drugs-beta")
+    logger.info(s"Writing outputs: ${outputs.mkString(",")}")
+
+    val outputConfs =
+      Helpers.generateDefaultIoOutputConfiguration(outputs: _*)(context.configuration)
+
+    val outputDFs = (outputs zip Seq(drugDf)).toMap
+
+    Helpers.writeTo(outputConfs, outputDFs)
+  }
+
+  private def printDetailedLogging(molecule: DataFrame,
+                                   indication: DataFrame,
+                                   mechanismOfAction: DataFrame,
+                                   targetsAndDiseases: DataFrame): Unit = {
+    val columnString: DataFrame => String = _.columns.mkString("Columns: [", ",", "]")
```

Comment:
@JarrodBaker I think it could be better putting the code of this function inside a logger whenEnabled fcuntion as the whole content is purely defined for logging and better to be sure nothing is computed if tracing is not enabled. something similar to 
```scala
    logger.whenTraceEnabled {
          val columnString: DataFrame => String = _.columns.mkString("Columns: [", ",", "]")
             logger.trace(s"""Intermediate dataframes:
             Columns:
             \n\t Molecule: ${columnString(molecule)},
             \n\t Indications: ${columnString(indication)},
             \n\t Mechanisms: ${columnString(mechanismOfAction)},
             \n\t Linkages: ${columnString(targetsAndDiseases)}
             Row counts:
             \n\t Molecule: ${molecule.count},
             \n\t Indications: ${indication.count},
             \n\t Mechanisms: ${mechanismOfAction.count},
             \n\t Linkages: ${targetsAndDiseases.count}
             """)
    }